enterprise/backup/src/main/java/org/neo4j/backup/BackupStrategyWrapper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+import java.io.File;
+import java.nio.file.Path;
+
+import org.neo4j.commandline.admin.CommandFailed;
+
+import static org.neo4j.backup.BackupState.failCritical;
+
+public class BackupStrategyWrapper
+{
+    private final BackupStrategyImplementationInterface backupStrategyImplementation;
+    private final BackupCopyService backupCopyService;
+
+    public BackupStrategyWrapper( BackupStrategyImplementationInterface backupStrategyImplementation, BackupCopyService backupCopyService )
+    {
+        this.backupStrategyImplementation = backupStrategyImplementation;
+        this.backupCopyService = backupCopyService;
+    }
+
+    public BackupState doBackup( OnlineBackupContext onlineBackupContext )
+    {
+        wrappedAsRuntimeException( backupStrategyImplementation::init );
+        wrappedAsRuntimeException( backupStrategyImplementation::start );
+        BackupState state = performBackupWithoutLifecycle( onlineBackupContext );
```

Comment:
You could use `LifeSupport` to take a LifeCycle through its stages instead of doing this wrapping and manual calls, which also doesn't gracefully handles failures along the way.

Reply:
Addressed in existing commit